### PR TITLE
Add mutex to secure access to senders and subsribers maps

### PR DESCRIPTION
**Pull request by OptimumCode**

### CODE REVIEW

File: pkg/queue/rabbitmq/internal/event/router.go [hunk 1]
```diff
@@ -171,16 +181,44 @@ func (cer *CommonEventRouter) getSubscriber(pin string, subscriberType internal.
...
 	var result *CommonEventSender
-	if _, ok := cer.senders[pin]; ok {
-		result = cer.senders[pin]
+	result = cer.findSender(pin)
+	if result != nil {
 		return result
 	}
+
+	cer.mutex.Lock()
+	defer cer.mutex.Unlock()
```
Comment (Nikita-Smirnov-Exactpro) [inline pkg/queue/rabbitmq/internal/event/router.go:203]: Could we lock mutex before first check to execute check and create new sender atomically?

Comment (OptimumCode) [reply pkg/queue/rabbitmq/internal/event/router.go:203]: We call RLock in findSender\findSubscriber methods. Then if the sender\subscriber is not initialized we take Lock

Comment (Nikita-Smirnov-Exactpro) [reply pkg/queue/rabbitmq/internal/event/router.go:203]: I understand your idea:
* when sender / subscriber is already initialized than read lock is enough
* when sender / subscriber was initializing first time than we check existence under read lock. If entity isn't existed then double check and create new instance under write lock

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Comment by Nikita-Smirnov-Exactpro** [comment]
Question: why we use single mutex for sender and subscriber?

**Comment by OptimumCode** [comment]
> Question: why we use single mutex for sender and subscriber?

Just to avoid having several mutexes and mixing them out. In most cases, we will take only RLock and it will not block other threads

**Pull request merged**